lib/cloudapi2.js

```diff
@@ -2371,25 +2374,50 @@ function waitForVolumeStates(opts, callback) {
     assert.uuid(opts.id, 'opts.id');
     assert.arrayOfString(opts.states, 'opts.states');
     assert.optionalNumber(opts.interval, 'opts.interval');
+    assert.optionalNumber(opts.timeout, 'opts.timeout');
     assert.func(callback, 'callback');
+
     var interval = (opts.interval === undefined ? 1000 : opts.interval);
+    interval = Math.min(interval, opts.timeout);
```

Comment:
Good catch, it doesn't! I made `opts.timeout` optional after adding that code and didn't update it after the change.